pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -102,11 +117,30 @@ func (c *codec) Decode(data []byte, defaultGVK *unversioned.GroupVersionKind, in
 			}
 			return into, gvk, nil
 		}
+
+		// perform defaulting if requested
+		if c.defaulter != nil {
+			// create a copy to ensure defaulting is not applied to the original versioned objects
```

Comment:
I think I don't understand why we can't modify the original object. This has just been decoded here, so why we can't do it?